### ISSUE TITLE
Can't open link with any browser (sh: 1: w3m: not found)

### ISSUE THREAD

**Issue by XinTecK** [newsboat/newsboat#463]
**newsboat r2.14.1-dirty - https://newsboat.org/**:

Hey, I've got a problem, I can't open any link in browser by pressing 'o' in an article.
In my config file (`~/snap/newsboat/843/.newsboat/config`), i've put this line : 
`browser "w3m %u"` . When i stop newsboat, i can see this line in my terminal : `sh: 1: w3m: not found`. When i change my config line for `browser "firefox %u"`, it is the same message : `sh: 1: firefox: not found`.
This is really weird because when i put these following lines in my terminal, it works well : `sh`, then `w3m google.fr`.

Steps to reproduce the issue:

1. Go on an article

2. Try to opening it in browser by pressing 'o'

3. Nothing appens

**Other info you think is relevant**:

When I run Newsboat like that: `newsboat --log-file=newsboat.log --log-level=6`, try to do the same, and then look at "browser" in the log file, i've got this : 

`xinteck@xinteck-pc:~/snap/newsboat/843/.newsboat$ cat newsboat.log | grep "browser"`

[2019-03-08 01:04:46] DEBUG: ConfigParser::parse: tokenizing browser "w3m %u"
[2019-03-08 01:04:46] DEBUG: ConfigContainer::handle_action: action = browser, type = 4
   <media:description>Let's start hacking on the Surf browser (from the Suckless Project) to make it a bit more user-friendly. 
Puppeteer - Headless Chrome browser for scraping (instead of Phantom JS)
  /* will return true for any browser supporting the spec, Edge included */
  /* will return true for any browser supporting the spec, Edge included */
  /* will return true for any browser supporting the spec, Edge included */
  /* will return true for any browser supporting the spec, Edge included */
[2019-03-08 01:04:55] INFO: ItemViewFormAction::process_operation: starting browser
[2019-03-08 01:04:55] DEBUG: View::open_in_browser: running `w3m 'http://www.developpez.com/actu/250074/Firefox-67-Mozilla-prevoit-d-ajouter-une-nouvelle-technique-anti-fingerprinting-appelee-letterboxing-empruntee-au-navigateur-Tor/''`

Thanks in advance !

**Comment by Minoru** [comment]
Hi! With Snap, the only browser you can use is `xdg-open`. It's security thing; Snap isolates applications to prevent them from doing something a user wouldn't want them to do. Set `browser` to `xdg-open %u`, and configure a URL handler through `xdg-settings`.

See #121 for some more details and context.

**Comment by XinTecK** [comment]
Hey ! When i set `browser ""xdg-open %u"` and start newsboat, i got the message : `sh: 1: xdg-open: Permission denied`. And i checked the issue #121 but i don't understand what do you mean by "configure a URL handler through `xdg-settings`"

Here is the log file : 

xinteck@xinteck-pc:~/snap/newsboat/843/.newsboat$ cat newsboat.log | grep "xdg"
[2019-03-08 09:56:45] DEBUG: ConfigParser::parse: tokenizing browser "xdg-open %u"
[2019-03-08 09:56:51] DEBUG: View::open_in_browser: running `xdg-open 'http://www.developpez.com/actu/250074/Firefox-67-Mozilla-prevoit-d-ajouter-une-nouvelle-technique-anti-fingerprinting-appelee-letterboxing-empruntee-au-navigateur-Tor/'`

Thanks for your help !

**Comment by Minoru** [comment]
> When i set browser ""xdg-open %u" and start newsboat, i got the message : sh: 1: xdg-open: Permission denied.

That might be a bug in Newsboat. Can you try `/bin/xdg-open %u` and `/usr/bin/xdg-open %u`, please?

>  don't understand what do you mean by "configure a URL handler through xdg-settings"

`xdg-open` doesn't know what browser to use to open URLs. A user has to run `xdg-settings` to tell it what browser to use. I never actually configured `xdg-open` myself, so I can't provide you with a worked example, but [the man page for `xdg-settings`](https://portland.freedesktop.org/doc/xdg-settings.html) has some example at the end.

**Comment by XinTecK** [comment]
Alright i see, thanks !

With `/bin/xdg-open %u` => `sh: 1: /bin/xdg-open: not found`
With `/usr/bin/xdg-open %u` => `sh: 1: /usr/bin/xdg-open: Permission denied`

Log file for `/usr/bin/xdg-open %u` : 

[2019-03-08 10:41:47] DEBUG: ConfigParser::parse: tokenizing browser "/usr/bin/xdg-open %u"
[2019-03-08 10:41:52] DEBUG: View::open_in_browser: running `/usr/bin/xdg-open 'http://www.developpez.com/actu/250074/Firefox-67-Mozilla-prevoit-d-ajouter-une-nouvelle-technique-anti-fingerprinting-appelee-letterboxing-empruntee-au-navigateur-Tor/''

**Comment by Minoru** [comment]
Reproduced in a virtual machine with Ubuntu 16.04.

Interestingly, if I build the Snap locally and install it in devmode, xdg-open works fine. Perhaps it's a feature of the devmode; need to learn more about that.

**Comment by XinTecK** [comment]
Uhm.. I see..

Alright, keep me informed !

**Comment by Minoru** [comment]
Turns out Snapcraft now requires us to explicitly ask for access to `xdg-open`. I updated the repository, and will push new releases to Snap Store once they're built.

Thank you for the report, @XinTecK! I wasn't aware of new requirements until you told me xdg-open no longer works; I wish Snapcraft did a better job informing us maintainers about the breaking changes. Right now their approach is to post everything on the forum that also gets user questions and such, which produces quite a bit of traffic (about 60 posts per week). I even saw [the one that announced the change](https://forum.snapcraft.io/t/the-desktop-interfaces/2042), but it didn't make it clear that this is a breaking change and I need to do something about it.
